### PR TITLE
DROOLS-3381: [DMN Designer] Do not add Header to nested ContextGrid

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-3381

This PR removes applicable column header meta-data when grids are nested.

This is required to correctly support keyboard navigation of nested grids; following review of https://github.com/kiegroup/kie-wb-common/pull/2200

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/556
- https://github.com/kiegroup/kie-wb-common/pull/2316

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 1]
```diff
@@ -1144,6 +1145,7 @@ public void test_function_definition_and_invocation() throws IOException {
     }
 
     @Test
+    @Ignore("Failing following https://issues.jboss.org/browse/DROOLS-3335")
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: @tarilabs This is a failing test following https://issues.jboss.org/browse/DROOLS-3335

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 2]
```diff
@@ -1228,6 +1230,7 @@ private void checkDecisionTableForPositiveOrNegative(final DMNRuntime runtime, D
 
     @Test
     @SuppressWarnings("unchecked")
+    @Ignore("Failing following https://issues.jboss.org/browse/DROOLS-3335")
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: @tarilabs This is a failing test following https://issues.jboss.org/browse/DROOLS-3335

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 3]
```diff
@@ -1268,6 +1271,7 @@ public void test_wrong_context() throws IOException {
...
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: @tarilabs This is a failing test following https://issues.jboss.org/browse/DROOLS-3335

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java [hunk 4]
```diff
@@ -38,7 +38,7 @@ public static void updateRowNumbers(final GridData uiModel,
         final Optional<GridColumn<?>> rowNumberColumn = uiModel
                 .getColumns()
                 .stream()
-                .filter(column -> column instanceof RowNumberColumn)
+                .filter(column -> column instanceof IsRowDragHandle)
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: @jomarko This is a duplication of what is in your PR (I needed it too, for these changes).

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridTheme.java [hunk 5]
```diff
@@ -87,7 +87,7 @@ public Rectangle getHeaderBackground(final GridColumn<?> column) {
             background.setFillColor(KIEColours.HEADER_BACKGROUND_LIGHT_BLUE);
         } else if (column instanceof LiteralExpressionColumn) {
             background.setFillColor(KIEColours.HEADER_BACKGROUND_LIGHT_BLUE);
-        } else if (column instanceof RowNumberColumn) {
+        } else if (column instanceof IsRowDragHandle) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridTheme.java]: @jomarko This is a duplication of what is in your PR (I needed it too, for these changes).

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java (hunk 4 above)
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: @manstis NP, I will fix when will rebase, thank you.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridTheme.java (hunk 5 above)
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridTheme.java]: @manstis NP, I will fix when will rebase, thank you.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@mbiarnes @tarilabs FYI this is another PR that failed as the related `appformer` PR appeared to not build the correct branch.. I've since kicked both builds off again (let's see if this fails again for the same reason)

**Comment by manstis** [comment]
@jomarko FYI I've pushed a new commit fixing selection of Header cells on nested grids. Now I only add the meta-data for _visible_ headers; it was possible that selection of a header cell (on row index 0) would cause execution of the incorrect action when the grid is nested (e.g. Function Grid has 2 header rows; one for the `HasExpression` - "Decision-1" and one for the Function to be executed. When not nested header row 0 is the `HasExpression` and selection of the cell should update the Properties Panel; however when the grid is nested header row 0 is the Function and selection should not update the Properties Panel).

**Comment by manstis** [comment]
@jomarko That my friend is up to you; I just wanted to let you know the PR has changed since you reviewed it (but I kept the change as a new commit).

**Comment by jomarko** [comment]
@manstis my friend, I will do re-review to have calm sleeping :)

**Comment by manstis** [comment]
@karreiro Could you please review at your convenience?

**Comment by manstis** [comment]
@karreiro you need to build the related appformer PR locally too.

**Pull request merged**